src/N3Lexer.js

```diff
@@ -471,7 +478,14 @@ export default class N3Lexer {
           }
           // Otherwise, tokenize as far as possible
           else {
-            this._input += data;
+            const dataAsString = String(data);
+            if (this._dataPrependTimes === 0) {
+              this._input = this._ignoreStartingBom(dataAsString);
+            }
+            else {
+              this._input += dataAsString;
+            }
+            this._dataPrependTimes++;
```

Comment:
Mmm can we do this more nicely? Like a boolean `this._bomMarkChecked` set by `_ignoreStartingBom`?

Reply:
Yes, we could do that. I think that's a matter of taste. I'll change it to a boolean.

Reply:
You're right, combined with all other suggestions, this makes a much prettier result.

Reply:
It might seem insignificant, but this function is called millions of times per second. Any write we don't have to do, has impact. (Plus at some point it would overflow to Infinity, that limit is easily reached when parsing a couple of 100 million triples.)